matplotlib2tikz/legend.py

```diff
@@ -181,6 +187,24 @@ def draw_legend(data, obj):
             alignment = None
             break
 
+    # Set color of lines in legend
+    temp = []
+    colors = set()
+    for handle in obj.legendHandles:
+        try:
+            data, legend_color, _ = mycol.mpl_color2xcolor(data,
+                                                           handle.get_color())
+            colors.add(legend_color)
+            temp.append('\\addlegendimage{no markers, %s}\n'
+                                         % legend_color)
+        except:
+            break
+
+    if colors.issubset(set(['black'])):
```

Comment:
maybe we should skip this, it makes it unnecessarily difficult. It only provides that in case all graph colors are black, addlegendimage entries will not be written in tex file, but the case that you have a legend with all graph colors are black is quite rare, I guess.